pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -239,9 +239,75 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Monitor the cloud-credential-operator-config configmap to check if operator is re-enabled.
+	// Check if attempts were made to delete the credentials requests when the operator was disabled.
+	// We have to have to reconcile all those credential requests.
+	configmapMapFn := handler.ToRequestsFunc(
+		func(a handler.MapObject) []reconcile.Request {
+			// Iterate all CredentialsRequests to determine if any of them have DeletionTimestamp set
+			log.Debug("checking for credentials requests marked for cleanup when operator was disabled")
+			crs := &minterv1.CredentialsRequestList{}
+			mgr.GetClient().List(context.TODO(), crs)
+			requests := []reconcile.Request{}
+			for _, cr := range crs.Items {
+				if cr.DeletionTimestamp != nil {
+					log.WithFields(log.Fields{
+						"cr": fmt.Sprintf("%s/%s", cr.Namespace, cr.Name),
+					}).Info("found credentials request marked for cleanup")
+					requests = append(requests, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Name:      cr.Name,
+							Namespace: cr.Namespace,
+						},
+					})
+				}
+			}
+			return requests
+		})
+
+	// Check if operator is re-enabled in cloud-credential-operator-config configmap
+	configMapPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isCloudCredOperatorRenabled(e)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+	}
+	err = c.Watch(
+		&source.Kind{Type: &corev1.ConfigMap{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: configmapMapFn,
+		},
+		configMapPredicate)
```

Comment:
I was worried we were now watching all ConfigMaps globally and would blow memory up in some clusters, but I can see we already have a configmap_controller that does the same, so our cache should be unchanged with your patch. :+1: